vignettes/available-sass-variables.Rmd

```diff
@@ -0,0 +1,251 @@
+      above_is_comment ~ lag(line),
+      TRUE ~ ""
+    ) %>% str_remove_all("\\/\\/") %>% str_trim(),
+    comment = ifelse(str_detect(comment, "stylelint-disable"), "", comment)
+  ) %>%
+  tidyr::fill(section, section_link) %>%
+  filter(!is.na(variable)) %>%
+  mutate(
+    variable = str_extract(line, "^\\$(.+)?(?=:)"),
+    value = str_extract(line, "(?<=:)(.+)?(?=;)") %>% str_remove(fixed("!default")) %>% str_trim(),
+    value_is_function = str_detect(value, fixed("()")) | str_detect(value, "\\($"),
+    # map-get function is used to reference a map built earlier so make sure not
+    # to link to it as it's not available in the table
+    value = str_replace_all(
+      value,
+      "(?<!map-get\\()(\\$[\\w|-]+)",
+      "<span class='dep-link'>\\1</span>"
+    ),
+    section = glue("Section: {section} <a href=\"{section_link}\"><i class=\"fas fa-external-link-alt\"></i></a>"),
+    declaration = glue("<a href=\"{link_to_pos_in_script(line_number)}\"><i class=\"fas fa-external-link-alt\"></i></a>")
+  ) %>%
+  filter(!value_is_function) %>%
+  select(
+    section,
+    variable,
+    value,
+    comment,
+    declaration
+  ) -> variables_df 
+```
+
+
+## Modifying Bootstrap Sass variable values
+
+Bootstrap uses `sass` to generate its final CSS. While a large portion of the customization you may want to do to your theme is easily done with the common arguments in `bs_theme()`, sometimes you may want finer-grain control. To enable this, `bslib` let's you change any of the variables used in the Bootstrap `sass` code. For instance, if you want to adjust the padding amount after each paragraph, that is available with the variable `paragraph-margin-bottom`, if we wanted to double the default amount (`1rem`) we could do so by simply passing the name of the variable to our `bs_theme()` call...
+
+```{r, eval = FALSE, echo = TRUE}
+theme <- bs_theme(
+  ...,
+  # Double space given after each paragraph
+  "paragraph-margin-bottom" = "2rem"
+)
+```
+
+
+### What's the "`$`" doing in front of the variables?
+You may notice in the table below that the variables are prefixed with a `$` this is a `sass` convention and you should omit the `$` in your calls to `bs_theme()`. However, if you wanted to reference another variable in your declaration of a value, such as making the `dark` variable equal to `grey-900` instead of `grey-800`, you _would_ need to use the `$` prefix when referring to the dependency. Also, you will need to tell `bslib` to place its definition later than `grey-900`'s with `bs_add_variables(..., .where = "declarations")`:
+
+```{r, eval = FALSE, echo = TRUE}
+theme <- bs_theme(
+  ...,
+  "paragraph-margin-bottom" = "2rem"
+) %>% 
+  # When referring to other variables, prefix with dollar-sign
+  bs_add_variables("dark" = "$gray-900", .where = "declarations")
+```
+
+For more details on this see the `sass` variables section of the [theming article.](https://rstudio.github.io/bslib/articles/theming.html#sass-variables-1)
```

Comment:
I think this should probably go below the table and we should consider removing the `$` in the variable column of the table (then, hopefully, users won't be so quick to wonder what the `$` is all about).

We'll have to, of course, explain why things like `$black` appear in the value column, as well as explain how to set a variable to default to another variable (e.g., `$dark: $gray-900`), but it seems like this should be more an after-thought (plus we could link to the other article, which also explains this)

Reply:
I agree. I think especially the readers of this article will understand the need to prefix variable names with some sort of identifier for sass.